modal-api.md

```diff
@@ -0,0 +1,58 @@
+# Design ideas for modal dialogs
+
+## The simple approach
+
+Several of the UX modal designs are pretty simple, consisting of a title, some text and some buttons (examples). These prompt-like intefaces could be supported with a cross-frame message with the format:
+
+```TypeScript
+interface Prompt = {
+  title : string,
+  message : string,
+  button : array<Button>
```

Comment:
We may have to account for the UX guidelines seperation of the left-side and right-side buttons.  Unless we assume the first or last is the right side.